File: db4o.net/Db4objects.Db4o.CS/Db4objects.Db4o.CS/CS/Internal/Messages/MObjectSet.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.CS.Internal;
using Db4objects.Db4o.CS.Internal.Messages;
using Db4objects.Db4o.Internal.Query.Result;

namespace Db4objects.Db4o.CS.Internal.Messages
{
	/// <exclude></exclude>
	public abstract class MObjectSet : MsgD
	{
		protected virtual AbstractQueryResult QueryResult(int queryResultID)
		{
			return Stub(queryResultID).QueryResult();
		}

		protected virtual LazyClientObjectSetStub Stub(int queryResultID)
		{
			IServerMessageDispatcher serverThread = ServerMessageDispatcher();
			return serverThread.QueryResultForID(queryResultID);
		}
	}
}

File: db4o.cs/src/com/db4o/cs/internal/messages/MObjectSet.java
/* Copyright (C) 2006  Versant Inc.  http://www.db4o.com */

package com.db4o.cs.internal.messages;

import com.db4o.cs.internal.*;
import com.db4o.internal.query.result.*;


/**
 * @exclude
 */
public abstract class MObjectSet extends MsgD {
	
	protected AbstractQueryResult queryResult(int queryResultID){
		return stub(queryResultID).queryResult();
	}

	protected LazyClientObjectSetStub stub(int queryResultID) {
		ServerMessageDispatcher serverThread = serverMessageDispatcher();
		return serverThread.queryResultForID(queryResultID);
	}

}
